### PR TITLE
Stop using SlurmGCP `nodeset` module.

### PR DESCRIPTION
**Pull request by mr0re1**
Stop using SlurmgGCP `nodeset` module, move all logic to this repo.

* No changes to behavior;
* Remove type constraints on `partition.variable`.

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf [hunk 1]
```diff
@@ -66,23 +66,22 @@ module "slurm_nodeset_template" {
...
 
-module "slurm_nodeset" {
-  source   = "github.com/GoogleCloudPlatform/slurm-gcp.git//terraform/slurm_cluster/modules/slurm_nodeset?ref=6.5.13"
-  for_each = local.nodeset_map
-
-  instance_template_self_link = module.slurm_nodeset_template[each.key].self_link
-
-  enable_placement       = each.value.enable_placement
-  maintenance_interval   = each.value.maintenance_interval
-  network_tier           = each.value.network_tier
```
Comment (harshthakkar01) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf:77]: Is network_tier not needed ?

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf [hunk 2]
```diff
@@ -66,23 +66,22 @@ module "slurm_nodeset_template" {
...
-  node_count_dynamic_max = each.value.node_count_dynamic_max
-  node_count_static      = each.value.node_count_static
-  nodeset_name           = each.value.nodeset_name
-  node_conf              = each.value.node_conf
-  subnetwork_self_link   = each.value.subnetwork_self_link
```
Comment (harshthakkar01) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf:82]: Are we moving from subnetwork_self_link to subnetwork  ?

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf (hunk 1 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf:77]: It's not used by scripts.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf (hunk 2 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/partition.tf:82]: No, I'm just matching what script expects from stored config.

**Review by harshthakkar01 — APPROVED** [review]

**Pull request merged**